module_fcst_grid_comp.F90

```diff
@@ -455,157 +631,101 @@ subroutine fcst_initialize(fcst_comp, importState, exportState, clock, rc)
           if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          if (ESMF_LogFoundError(rcToCheck=urc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__, rcToReturn=rc)) return
 
-          if ( npes(1) == TileLayout(1) * TileLayout(2) * 6 ) then
-            ! global-nesting
-            nestRootPet = npes(1)
-            gridfile="grid.nest02.tile7.nc"
-          else if ( npes(1) == TileLayout(1) * TileLayout(2) ) then
-            ! regional-nesting
-            nestRootPet = npes(1)
-            gridfile="grid.nest02.tile2.nc"
-          else
-            call mpp_error(FATAL, 'Inconsistent nestRootPet and Atmos%layout')
-          endif
 
-          if (mype == nestRootPet) then
-            if (nestRootPet /= Atmos%pelist(1)) then
-              write(0,*)'error in fcst_initialize: nestRootPet /= Atmos%pelist(1)'
-              write(0,*)'error in fcst_initialize: nestRootPet = ',nestRootPet
-              write(0,*)'error in fcst_initialize: Atmos%pelist(1) = ',Atmos%pelist(1)
-              ESMF_ERR_ABORT(100)
-            endif
-          endif
+          deallocate(petListNest)
 
-          ! nest rootPet shares peList with others
-          if (mype == nestRootPet) peListSize(1) = size(Atmos%pelist)
-          call ESMF_VMBroadcast(vm, bcstData=peListSize, count=1, rootPet=nestRootPet, rc=rc)
-          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+        end if
 
-          ! nest rootPet shares layout with others
-          if (mype == nestRootPet) regDecomp = Atmos%layout
-          call ESMF_VMBroadcast(vm, bcstData=regDecomp, count=2, rootPet=nestRootPet, rc=rc)
-          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+        if (ESMF_GridCompIsPetLocal(fcstGridComp(n), rc=rc)) then
+          call ESMF_GridCompGet(fcstGridComp(n), grid=fcstGrid(n), rc=rc); ESMF_ERR_ABORT(rc)
 
-          ! prepare petMap variable
-          allocate(petMap(peListSize(1)))
-          if (mype == nestRootPet) petMap = Atmos%pelist
-          ! do the actual broadcast of the petMap
-          call ESMF_VMBroadcast(vm, bcstData=petMap, count=peListSize(1), rootPet=nestRootPet, rc=rc)
-          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          call ESMF_GridAddCoord(fcstGrid(n), staggerLoc=ESMF_STAGGERLOC_CENTER, rc=rc); ESMF_ERR_ABORT(rc)
+          call ESMF_GridAddCoord(fcstGrid(n), staggerLoc=ESMF_STAGGERLOC_CORNER, rc=rc); ESMF_ERR_ABORT(rc)
 
-          ! create the DELayout that maps DEs to the PETs in the petMap
-          delayout = ESMF_DELayoutCreate(petMap=petMap, rc=rc)
-          if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          ! define "center" coordinate values
```

Comment:
It seems cleaner to stay with radians, and not to convert to degrees. As discussed previously, and shown by Dusan's test runs, we already are seeing bit-for-bit changes due to changing over to the FCST coords (instead of reading from file for global). Might as well deal with bit changes once, remove dtor variable and be done with it.

Reply:
This is obviously the wrong section of the code showing up in the comment. I had highlighted where the conversions are being computed, filling in the coord values.

Reply:
@junwang-noaa what do you think? Personally I much rather see the ESMF fcstGrid defined **exactly** as the model uses it, No conversion at all. We have ESMF grid as R8, so we can capture FV3's geometry arrays precisely, no matter how FV3 is compiled. I understand that the fcstGrid is in degrees, and that is fine, it just stays that way. ESMF will do the regridding either way. Less convenversions on the user level just seems clearer. And since we have bit changes no matter what, we might as well do it as clean in the PR as we can.

Reply:
I want to confirm that we are talking about line 706, 711,717 and 722. I think it' good to keep the coordinates same as the model especially those grids are also used in other places such as coupling. We may also need to write out the coordinates in radians not in degrees for consistency in netcdf files.  We can make the changes in next PR.

Reply:
So remove the conversion here, and get rid of the 'dtor' variable in this PR. Then deal with output changes you mention in a future PR.

Reply:
If we remove conversion here, conversion from rad to deg must be added to module_write_netcdf, and somewhere in ioCompRun or ESMFproto_FieldBundleWrite.

Reply:
Since this PR does not change model output results except adding multiple domain output files for HAFS tests. I think it's better we keep the code changes in this PR and have a separate PR with radian coordinates for both forecast grids, write grids and coordinates in output files .

Reply:
@theurich That is correct, thanks. Dusan made the code changes, please review again. Thanks

Reply:
That looks good to me now. Thank you!